dashboard-api/internal/account.go

```diff
@@ -0,0 +1,85 @@
+package studiesmanager
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+)
+
+type AccountRepository interface {
+	Create(ctx context.Context, a Account) error
+}
+
+var ErrAccountAlreadyExists = errors.New("Account Already Exists")
+
+//go:generate mockery --case=snake --outpkg=storagemocks
+//--output=platform/storage/storagemocks --name=AccountRepository
+
+type Account interface {
+	GetName() string
+	GetUserID() string
+	GetAuthType() string
+	MarshalCredentials() (string, error)
+}
+
+type AccountBase struct {
+	ID        string    `json:"id" `
+	UserID    string    `json:"userId"`
+	AuthType  string    `json:"authType" validate:"required"`
+	Name      string    `json:"name" validate:"required"`
+	CreatedAt time.Time `json:"createdAt"`
+}
+
+func (a AccountBase) GetName() string {
```

Comment:
Are these getters really necessary? Feels Java.

Reply:
Yes, as the type will change based on i.e `Typeform` account and/or `Fly` account so we use these on the interface in order to maintain type integrity (i.e we know whatever we pass into the create function will definitely have this and wont cause a nil pointer)

Reply:
Im actually completley wrong, I changed it form using embeded structs and just used an interface on the `ConnectedAccount` which seems to remove the need for getters